orderbook/src/orderbook.rs

```diff
@@ -203,6 +206,23 @@ impl Orderbook {
     }
 }
 
+async fn filter_unsupported_tokens(
+    mut orders: Vec<Order>,
+    bad_token: &dyn BadTokenDetecting,
+) -> Result<Vec<Order>> {
+    let mut index = 0;
+    'outer: while index < orders.len() {
+        for token in orders[index].order_creation.token_pair().unwrap() {
+            if !bad_token.detect(token).await?.is_good() {
```

Comment:
The fancy double loop is needed because detect is async and so using things like iter.any doesn't work? Maybe add a comment or consideration if this should run in parallel in the future?